src/linagora.esn.contact/app/app.controllers.js

```diff
@@ -35,6 +35,14 @@ angular.module('linagora.esn.contact')
     $scope.contact = sharedContactDataService.contact;
     $scope.addressbookPath = '/addressbooks/' + $scope.bookId + '/' + $scope.bookName + '.json';
 
+    function _hideModal() {
+      $scope.$hide();
+    }
+    $scope.$on('$stateChangeStart', function(event, toState) {
+      if (toState.name !== 'contact.addressbooks.new') {
+        _hideModal();
+      }
+    });
```

Comment:
this listens to every state change in the application.

you can simply close the modal when leaving the state by adding 
```js
          onExit: function(modalInstance) {
            modalInstance.hide();
          }
```
to the state